pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -1534,6 +1581,34 @@ func mergeDashbuilder(service kubernetes.PlatformService, cr *api.KieApp, env ap
 	return env, nil
 }
 
+func overrideKafkaTopicsEnv(cr *api.KieApp, env *api.Environment) {
+	var dataSet []string
+	if cr.Status.Applied.Objects.Servers[0].Kafka != nil {
+		if cr.Status.Applied.Objects.Servers != nil {
```

Comment:
shouldn't this if condition be before? than kafka struct test?

Reply:
yes